tests/src/lib.rs

```diff
@@ -4,6 +4,85 @@ use generic_ec::Curve;
 use rand::RngCore;
 use serde_json::{Map, Value};
 
+/// Wraps a sink to buffer the messages. Used in [`buffer_outgoing`]
+#[pin_project::pin_project]
+pub struct BufferedSink<M, Inner> {
+    #[pin]
+    messages: std::collections::VecDeque<M>,
+    #[pin]
+    inner: Inner,
+}
+type BufferedDelivery<M, D> = (
+    <D as round_based::Delivery<M>>::Receive,
+    BufferedSink<round_based::Outgoing<M>, <D as round_based::Delivery<M>>::Send>,
+);
+
+impl<M: Unpin, Inner: futures::Sink<M>> futures::Sink<M> for BufferedSink<M, Inner> {
+    type Error = Inner::Error;
+
+    fn poll_ready(
+        self: std::pin::Pin<&mut Self>,
+        _cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Result<(), Self::Error>> {
+        // Always ready to buffer
+        std::task::Poll::Ready(Ok(()))
+    }
+
+    fn start_send(self: std::pin::Pin<&mut Self>, item: M) -> Result<(), Self::Error> {
+        self.project().messages.get_mut().push_back(item);
+        Ok(())
+    }
+
+    fn poll_flush(
+        mut self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Result<(), Self::Error>> {
+        // Feed all buffered messages one by one
+        while !self.messages.is_empty() {
+            let mut projection = self.as_mut().project();
+            let mut inner = projection.inner;
+            // In case the inner sink wasn't ready, this method will be retried.
+            // We rely on this and don't modify any internal state before this
+            // point
+            std::task::ready!(inner.as_mut().poll_ready(cx))?;
+            if let Some(item) = projection.messages.pop_front() {
+                inner.as_mut().start_send(item)?;
+            }
+        }
+        self.project().inner.poll_flush(cx)
+    }
+
+    fn poll_close(
+        self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Result<(), Self::Error>> {
+        self.project().inner.poll_close(cx)
+    }
+}
+
+/// Map a party to add explicit buffering to outgoing messages in case the
+/// underlying sink doesn't do buffering. This is useful to test that we don't
+/// forget to flush the sinks in our protocol implementations.
+pub fn buffer_outgoing<M, D, R>(
+    party: round_based::MpcParty<M, D, R>,
+) -> round_based::MpcParty<M, BufferedDelivery<M, D>, R>
```

Comment:
I dunno, sounds a bit elaborate. Maybe this?

 ```suggestion
/// Modifies `Delivery` implementation of the `MpcParty` so sink of outgoing messages
/// buffers messages to be sent until `flush` is called.
///
/// cggmp21 protocol implementation often intentionally delay the moment the `flush` is called
/// to optimize network delays and let delivery implementation parallelize message sending.
/// However, we might accidentally forget to call `flush()` in the end of the round. By calling this
/// method in the tests and buffering all outgoing messages until `flush` is called we make sure
/// that we did not forget to call `flush`: otherwise tests will deadlock. 
pub fn buffer_outgoing<M, D, R>(
    party: round_based::MpcParty<M, D, R>,
) -> round_based::MpcParty<M, BufferedDelivery<M, D>, R>
```